styx-cli/src/main/java/com/spotify/styx/cli/JsonCliOutput.java

```diff
@@ -48,7 +51,8 @@ private static void printJson(Object object) {
 
   @Override
   public void printStates(RunStateDataPayload runStateDataPayload) {
-    printJson(CliUtil.groupStates(runStateDataPayload.activeStates()));
+    printJson(CliUtil.groupStates(runStateDataPayload.activeStates()).entrySet().stream()
+        .collect(toMap(e -> e.getKey().toKey(), Entry::getValue)));
```

Comment:
Slightly confused as to what is going on here (but I think it's just my lack of Java knowledge). What is happening in `.collect`? Is that where we transform `"WorkflowId{componentId=foo-component, id=foo-workflow}"` to `"foo-component#foo-workflow"`?

Reply:
@Zatine You just happened to discover the dark side of Java lambda: it is soooo verbose. This essentially creates a map keyed on `e.getKey().toKey()`, mapped to value `e.getValue()`.

Reply:
Ah, thanks. My mistake was looking at the `getKey().toKey()` in general contexts online when it was actually methods on the WorkflowInstance. It's get it now!